lib/temp.rb

```diff
@@ -0,0 +1,34 @@
+
+def temp_exists?(name)
+  File.exists?(temp_join(name))
+end
+
+def temp_join(name)
```

Comment:
You may be able to avoid a few of these methods with Ruby's [Tempfile](http://ruby-doc.org/stdlib-2.6.3/libdoc/tempfile/rdoc/Tempfile.html) class.

Reply:
Decided to stick with `File` for now because I'm not sure if the memoization would work right with `Tempfile`s out-of-scope semantics.